plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -3524,6 +3524,46 @@ eosio::chain::backing_store_type read_only::get_backing_store() const {
    return kv_database.get_backing_store();
 }
 
+read_only::get_all_accounts_result
+read_only::get_all_accounts( const get_all_accounts_params& params ) const
+{
+   get_all_accounts_result result;
+
+   using acct_obj_idx_type = chainbase::get_index_type<chain::account_object>::type;
+   const auto& accts = db.db().get_index<acct_obj_idx_type >().indices().get<chain::by_name>();
+
+   auto cur_time = fc::time_point::now();
+   auto end_time = cur_time + fc::microseconds(1000 * 10); /// 10ms max time
+   uint32_t start_idx = params.page * params.page_size;
+   
+   
+   auto end_itr = accts.end();
+   auto itr = accts.begin();
+   auto d = std::distance(itr, end_itr);
+   if (start_idx < d) {
+      std::advance(itr, start_idx);
+   }
+   else {  
+      result.more = false;
+      return result;
+   }
+
+   while(cur_time <= end_time
+         && result.accounts.size() < params.page_size
+         && itr != end_itr)
+   { 
+      auto a = *itr;
+      result.accounts.push_back({a.name, a.creation_date});
+      itr++;
+      
+      cur_time = fc::time_point::now();
+   }   
+
+   result.more = (itr != end_itr && result.accounts.size() < params.page_size);
```

Comment:
Sorry, I was not clear. Please follow example of something like `get_activated_protocol_features` where more indicates where to start on the next call.

Reply:
@heifner See comment above, I am now using lower_bound and upper_bound, together with 'limit' in params, and result has optional 'more' field, so clients will know if there are more results that need to be fetched.